### PR TITLE
fix for sporadic porch crash during new package revision

### PR DESCRIPTION
**Pull request by sriharshabm**
Current code in package variant controller was assuming that "status.upstreamLock.Git.Ref" in PackageRevision resource will always having proper value. But status.upstreamLock field in PackageRevision could be nil in initial transition stage. Due to which Porch was crashing. Issue is fixed as part of this PR

### CODE REVIEW

**Comment by liamfallon** [comment]
Please also check the dates on the copyright notices on the top of the go files.

**Review by sriharshabm — COMMENTED** [review]
updated warning message

**Comment by kispaljr** [comment]
(CLA signing is in progress)

File: controllers/packagevariants/pkg/controllers/packagevariant/injection.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2023 The kpt and Nephio Authors
+// Copyright 2024 The kpt and Nephio Authors
```
Comment (liamfallon) [inline controllers/packagevariants/pkg/controllers/packagevariant/injection.go:1]:
```suggestion
// Copyright 2023-2024 The kpt and Nephio Authors
```

File: controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2022 The kpt and Nephio Authors
+// Copyright 2024 The kpt and Nephio Authors
```
Comment (liamfallon) [inline controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go:1]:
```suggestion
// Copyright 2023-2024 The kpt and Nephio Authors
```

**Review by liamfallon — COMMENTED** [review]
Looking good except for the pedantic date comment :-)

**Review by liamfallon — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**